packages/client/mutations/useAcceptRequestToJoinDomainMutation.ts

```diff
@@ -0,0 +1,71 @@
+import graphql from 'babel-plugin-relay/macro'
+import {useMutation, UseMutationConfig} from 'react-relay'
+import useAtmosphere from '../hooks/useAtmosphere'
+import {useAcceptRequestToJoinDomainMutation as TAcceptRequestToJoinDomainMutation} from '../__generated__/useAcceptRequestToJoinDomainMutation.graphql'
+import SendClientSegmentEventMutation from './SendClientSegmentEventMutation'
+
+graphql`
+  fragment useAcceptRequestToJoinDomainMutation_success on AcceptRequestToJoinDomainSuccess
+  @argumentDefinitions(requestId: {type: "ID!"}) {
+    viewer {
+      ...ReviewRequestToJoinOrgModal_viewer @arguments(requestId: $requestId)
+    }
+  }
+`
+
+const mutation = graphql`
+  mutation useAcceptRequestToJoinDomainMutation($requestId: ID!, $teamIds: [ID!]!) {
+    acceptRequestToJoinDomain(requestId: $requestId, teamIds: $teamIds) {
+      ... on ErrorPayload {
+        error {
+          message
+        }
+      }
+      ...useAcceptRequestToJoinDomainMutation_success @arguments(requestId: $requestId)
+    }
+  }
+`
+type Handlers = {
+  onSuccess?: () => void
+  onError?: () => void
+}
+const useAcceptRequestToJoinDomainMutation = () => {
+  const [commit, submitting] = useMutation<TAcceptRequestToJoinDomainMutation>(mutation)
+  const atmosphere = useAtmosphere()
+  const execute = (
+    config: UseMutationConfig<TAcceptRequestToJoinDomainMutation>,
+    handlers?: Handlers
+  ) => {
+    const {variables} = config
+    return commit({
+      onCompleted: (res) => {
+        const error = res.acceptRequestToJoinDomain.error
+
+        if (!error) {
+          atmosphere.eventEmitter.emit('addSnackbar', {
+            message: '🎉 Success! User added',
+            autoDismiss: 5,
+            key: 'acceptRequestToJoinDomainSuccess'
+          })
+          SendClientSegmentEventMutation(atmosphere, 'Join Request Reviewed', {
+            action: 'accept',
+            teamIds: variables.teamIds
+          })
+          handlers?.onSuccess && handlers.onSuccess()
+        } else {
+          atmosphere.eventEmitter.emit('addSnackbar', {
+            message: error.message,
+            autoDismiss: 5,
+            key: 'acceptRequestToJoinDomainError'
+          })
+          handlers?.onError && handlers.onError()
+        }
+      },
+      // allow components to override default handlers
+      ...config
+    })
+  }
+  return [execute, submitting] as const
```

Comment:
+1 I'm not certain having these in an array rather than an object is the best way forward, but happy to see how it evolves first.

Reply:
Merging this for now, some discussion about possible improvement here #8299